Services/DeviceModels.cs

```diff
@@ -153,5 +184,36 @@ private bool CheckStockDeviceModelExistence(string id)
             return this.stockDeviceModels.GetList()
                 .Any(model => id.Equals(model.Id, StringComparison.InvariantCultureIgnoreCase));
         }
+
+        /// <summary>
+        /// Transforms property names from Json Object to "xxx.yyy.zzz" format
+        /// </summary>
+        private void PreparePropertyNames(HashSet<string> set, object obj, string prefix)
+        {
+            /* Sample conversion:
+             * from -> Foo : {
+             *                  Bar : Properties
+             *               }
+             *          
+             * to -> Foo.Bar.Properties
+             */
+            if (obj is JValue)
+            {
+                set.Add(prefix);
+                return;
+            }
+
+            if (obj is bool || obj is string || double.TryParse(obj.ToString(), out _))
+            {
+                set.Add(prefix);
+                return;
+            }
+
+            foreach (var item in (obj as JToken).Values())
+            {
+                var path = item.Path;
+                this.PreparePropertyNames(set, item, $"{prefix}.{(path.Contains(".") ? path.Substring(path.LastIndexOf('.') + 1) : path)}");
```

Comment:
please fix indentation

Reply:
@hathind-ms indentation looks right. Am I missing something?

Reply:
weird, must be screen resolution, it looked wrap on my laptop.